go/teams/service_helper.go

```diff
@@ -661,20 +661,38 @@ func AcceptInvite(ctx context.Context, g *libkb.GlobalContext, token string) err
 	return err
 }
 
-func ParseAndAcceptSeitanToken(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
+func parseAndAcceptSeitanTokenV1(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
 	seitan, err := GenerateIKeyFromString(tok)
 	if err != nil {
 		g.Log.CDebugf(ctx, "GenerateIKeyFromString error: %s", err)
 		g.Log.CDebugf(ctx, "returning TeamInviteBadToken instead")
 		return false, libkb.TeamInviteBadTokenError{}
 	}
-	g.Log.CDebugf(ctx, "found seitan token")
 	err = AcceptSeitan(ctx, g, seitan)
+	return true, err
+}
+
+func parseAndAcceptSeitanTokenV2(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
+	seitan, err := GenerateIKeyV2FromString(tok)
 	if err != nil {
-		return true, err
+		g.Log.CDebugf(ctx, "GenerateIKeyV2FromString error: %s", err)
+		g.Log.CDebugf(ctx, "returning TeamInviteBadToken instead")
+		return false, libkb.TeamInviteBadTokenError{}
 	}
+	err = AcceptSeitanV2(ctx, g, seitan)
+	return true, err
+
+}
 
-	return true, nil
+func ParseAndAcceptSeitanToken(ctx context.Context, g *libkb.GlobalContext, tok string) (wasSeitan bool, err error) {
+	wasSeitan, err = parseAndAcceptSeitanTokenV1(ctx, g, tok)
+	if err != nil {
+		wasSeitan, err = parseAndAcceptSeitanTokenV2(ctx, g, tok)
+	}
+	if wasSeitan {
+		g.Log.CDebugf(ctx, "found seitan token")
```

Comment:
yeah just be super super careful to not regurgitate the seitan token back into the output.

Reply:
yup i went over the logs (@zapu suggestion) and cleaned up anything i saw